basic-examples/expressions/workflow.yaml

```diff
@@ -1,25 +1,49 @@
----
 version: v0.2.0
 input:
   root: RootObject
   objects:
     RootObject:
       id: RootObject
-      properties: {}
+      properties:
+        name:
+          display:
+            description: Just a name
+            name: Name
+          required: true
+          type:
+            type_id: string
+        int_value:
+          display:
+            description: Just an integer value
+            name: Integer Value
+          required: true
+          type:
+            type_id: integer
+        float_value:
+          display:
+            description: Just a float value
+            name: Float Value
+          required: true
+          type:
+            type_id: float
+
 steps:
-  expression_loop:
-    kind: foreach
-    workflow: subworkflow.yaml
-    items:
-      - name: !expr '"Here''s an apostrophe and \"embedded quotes\"."'
-      - name: !expr "'Here\\'s an apostrophe and \"embedded quotes\".'"
-      - name: !expr |-
-          'Here\'s an apostrophe and "embedded quotes".'
-      - name: !expr |-
-          "Here's an apostrophe and \"embedded quotes\"."
-      - name: !expr '`Here''s an apostrophe and "embedded quotes".`'
-      - name: !expr |-
-          `Here's an apostrophe and "embedded quotes".`
+  example1:
+    plugin:
+      deployment_type: image
+      src: quay.io/arcalot/arcaflow-plugin-template-python:0.4.0
+    input:
+      #Expression with concatenation and string literal
+      name: !expr $.input.name +  ' - \"The Noble Workflow Engine\"'

```

Comment:
Consider using back-quotes (see the old code line 20) so that we can avoid having to escape the double quotes.

Reply:
I'm purposefully escaping the double quotes to show how they are used in the string literal.

Reply:
OK, but, really...escapes should be more or less the last resort, not the first one.

Reply:
This is literally one of the quoting examples you are concerned about losing in a separate comment. Arguably almost none of those existing examples are very practical, but I chose to keep one that was unambiguous.

Reply:
Part of the power of those examples is that they appear together as a set.  Picking out one of them dilutes the demonstration of how to code strings.

As a matter of fact, none of those examples are ambiguous:  they all encode the same value using different mechanisms, each of which is available to our users.

The point of that set is that, yes, one can use escape characters if that's what you're comfortable with because you're coming from a language or situation where you were forced to do so, but there are also alternatives which allow you to use quotation marks as literal characters without having to resort to inserting other, special characters into your string.  Users should be free to select the approach which best suits their needs.

The question is, what should we be presenting to our users?  Is the presence of backslashes in the string going to look weird or familiar?  Will it (and the quotes) improve the example or detract from it?  Would it be better to make the example more focused (e.g., by avoiding the quotation marks) or to try to make it encompass as many concepts as we can fit into the limited space (economy of scale and all that)?

I thought that including the escapes was, in the balance, not worthwhile.  If we really want to demonstrate how to address the challenge of embedding quotation marks, I think we should employ an option which doesn't insert extra magic inside the string.  Or, better still, we should have an example which compares and contrasts all the approaches to quoting strings....

Reply:
In terms of demonstrating the escape characters, I don't see a great other example that will communicate anything practical here except the quote escaping. *Maybe* I could could do something with a newline or a tab...

https://arcalot.io/arcaflow/workflows/expressions/#string-values

Reply:
Right, but first we should answer the question of whether we want to touch on the question of escape characters at all in this particular example.  This is nominally a "basic" example -- are character escapes _basic usage_?  This example would be just as valid without including embedded quotes in the to-be-concatenated string.

It seems like we should have another example centered on the interplay between YAML's processing and Arca's expression language and how to get your strings to look the way you want them to, since that's something that more advanced users are seemingly bound to face.  And, while `\"` is certainly a comfortable solution if you are used to C or Bash programming, we now have various alternatives, including raw strings, which deliver you from having to use metacharacters, and it seems like we might want to suggest those as a solution, rather then trotting out the old tried-and-true ugliness.